pfcpiface/pfcpiface.go

```diff
@@ -422,59 +448,88 @@ func handleSessionModificationRequest(upf *upf, msg message.Message, addr net.Ad
 				farID, err := ie1.FARID()
 				if err != nil {
 					log.Println("Unable to find updateFAR's FAR ID!")
-				}
-
-				/* Read UpdateFAR from payload */
-				ies2, err := ie1.UpdateForwardingParameters()
-				if err != nil {
-					log.Println("Unable to find UpdateForwardingParameters!")
 				} else {
-					for _, ie2 := range ies2 {
-						switch ie2.Type {
-						case ie.OuterHeaderCreation:
-							outerheadercreationfields, err := ie2.OuterHeaderCreation()
-							if err != nil {
-								log.Println("Unable to parse OuterHeaderCreationFields!")
-							} else {
-								eNBTeid := outerheadercreationfields.TEID
-								eNBIP := outerheadercreationfields.IPv4Address
-								s1uIP4 := upf.n3IP
-								far := far{
-									farID:       uint8(farID),  // farID currently being truncated to uint8 <--- FIXIT/TODO/XXX
-									fseID:       uint32(fseid), // fseID currently being truncated to uint32 <--- FIXIT/TODO/XXX
-									action:      farForwardD,
-									tunnelType:  0x1,
-									s1uIP:       ip2int(s1uIP4),
-									eNBIP:       ip2int(eNBIP),
-									eNBTeid:     eNBTeid,
-									UDPGTPUPort: udpGTPUPort,
+					var eNBTeid uint32
+					eNBIP := uint32(0)
+					n6IP4 := uint32(0)
+					tunnelType := uint8(0)
+					/* Read UpdateFAR from payload */
+					ies2, err := ie1.UpdateForwardingParameters()
+					if err != nil {
+						log.Println("Unable to find UpdateForwardingParameters!")
+					} else {
+						for _, ie2 := range ies2 {
+							switch ie2.Type {
+							case ie.OuterHeaderCreation:
+								outerheadercreationfields, err := ie2.OuterHeaderCreation()
+								if err != nil {
+									log.Println("Unable to parse OuterHeaderCreationFields!")
+								} else {
+									eNBTeid = outerheadercreationfields.TEID
+									eNBIP = ip2int(outerheadercreationfields.IPv4Address)
+									n6IP4 = ip2int(upf.n3IP)
+									tunnelType = uint8(1)
 								}
-								/* create context */
-								ctx, cancel := context.WithTimeout(context.Background(), Timeout)
-								defer cancel()
-								done := make(chan bool)
-								// pause daemon, and then insert FAR, finally resume
-								upf.pauseAll()
-								upf.addFAR(ctx, done, far)
-								upf.resumeAll()
+							case ie.DestinationInterface:
+								// Do nothing for the time being
 							}
-						case ie.DestinationInterface:
-							// Do nothing for the time being
 						}
```

Comment:
Use function on IE instead of us parsing
https://pkg.go.dev/github.com/wmnsk/go-pfcp@v0.0.1/ie?tab=doc#IE.OuterHeaderCreation
https://pkg.go.dev/github.com/wmnsk/go-pfcp@v0.0.1/ie?tab=doc#IE.DestinationInterface

Reply:
Same for other functions in pfcpiface.go wherever possible use library functions

Reply:
Closing this for the time being.